earthfile2llb/converter.go

```diff
@@ -2416,6 +2429,65 @@ func (c *Converter) internalRun(ctx context.Context, opts ConvertRunOpts) (pllb.
 	}
 }
 
+// detectAWSCredentials determines whether AWS envs are set and/or ~/.aws
+// exists. These are then used when fetching the secrets during a build.
+func (c *Converter) detectAWSCredentials(ctx context.Context) ([]llb.RunOption, []string, error) {
+
+	homeDir, err := os.UserHomeDir()
```

Comment:
I wonder if we should use https://github.com/earthly/earthly/blob/34320d5111d7488bee0ecbc961b4e9d51d289e06/util/fileutil/homedir.go#L53 instead?